### ISSUE TITLE
ListField of Document should return list of instances of that class

### ISSUE THREAD

**Issue by grundic** [prawn-cake/simple-models#13]
Right now ListField return list of dictionaries, which is incorrect:

``` python
from simplemodels.fields import CharField, ListField
from simplemodels.models import Document


class Comment(Document):
    body = CharField()


class User(Document):
    name = CharField()
    comments = ListField([Comment])


data = {
    'name': 'John Smith',
    'comments': [
        {'body': 'comment #1'},
        {'body': 'seconds comment'},
        {'body': 'one more comment'},
    ]
}

if __name__ == '__main__':
    user = User.create(data)
    print(user.name)
    for comment in user.comments:
        print(comment)
        assert isinstance(comment, Comment)  # actually, comment is dict
        assert (comment.__class__.__name__ == Comment.__name__), "{} != {}".format(
            comment.__class__.__name__, Comment.__name__)

>>> AssertionError: dict != Comment
```

**Comment by prawn-cake** [comment]
Confirm. It's because of [instancecheck](https://github.com/prawn-cake/simple-models/blob/master/simplemodels/models.py#L86) where I admited  that a dict having proper structure can be considered as an instance, but in fact yes, looks malfunctioning.

I feel that `ListField` interface becomes too ambigues, may be worth to restrict it to have one and only one data type inside a list

**Comment by grundic** [comment]
I would agree to have only one type inside ListField.
Otherwise it should be treated specially and each element in list should be checked for applicable for this or that type.

**Comment by grundic** [comment]
Also I would recommend to have a look at [jsonmapper](https://github.com/mgood/jsonmapper/blob/master/jsonmapper/__init__.py#L399) implementation of the ListField, as it's quite smart.

**Comment by prawn-cake** [comment]
Not sure about that complexity worth it and those inline internal type definitions looks very foreign to me, the interface not intuitive and it looks tedious to define such amount of nested structures. 
I'd better focus on "one-type" ListField feature

**Comment by grundic** [comment]
I'm fine with that. Just like to share the knowledge with you. I would also prefer "one-type" ListField.

**Comment by prawn-cake** [comment]
Fixed in the [v0.5.1](https://github.com/prawn-cake/simple-models/tree/0.5.1)
